### PR TITLE
parallelize cxxvertex creation

### PR DESCRIPTION
**Pull request by wkotlarski**
@Expander, are you sure your commit solves the problem? Because I'm still seeing it. Also, I think the code should be
```mathematica
   ParallelEvaluate[
      (BeginPackage[#];EndPackage[];)& /@ contextsToDistribute;
       (* prevent shdw warning with Susyno`LieGroups`M: *)
       Off[Remove::remal];
       Remove[Susyno`LieGroups`M];
       On[Remove::remal];,
      DistributedContexts->Automatic
   ];
```
The ```Remove[Susyno`LieGroups`M]``` should be run only once per kernel.

### CODE REVIEW

**Comment by Expander** [comment]
I've just checked it. It solves the `shdw` warning when generating the code in `models/`:
~~~
Creating a C++ code for decay amplitudes...
Launching kernels...
The creation of C++ code for decays took 12.2s

----------------------------------------------------------------------
Creating other observables
----------------------------------------------------------------------
Creating class for effective couplings ...
Creating class for observables ...
Creating EDM class ...
Creating l->l'A class ...
Creating FToFConversionInNucleus class ...
Creating FFMasslessV form factor class for other observables ...
~~~
However, I still see it when generating `test/test_SM_cxxdiagrams.meta`:
~~~
Read Vertex Lists
Launching kernels...

Part::pkspec1: The expression a cannot be used as a part specification.

Part::pkspec1: The expression a cannot be used as a part specification.

Part::pkspec1: The expression a cannot be used as a part specification.

General::stop: Further output of Part::pkspec1
     will be suppressed during this calculation.
From KernelObject[1, local]:
M::shdw: Symbol M appears in multiple contexts 
    {FlexibleSUSY`, Susyno`LieGroups`}; definitions in context FlexibleSUSY`
     may shadow or be shadowed by other definitions.
From KernelObject[2, local]:
M::shdw: Symbol M appears in multiple contexts 
    {FlexibleSUSY`, Susyno`LieGroups`}; definitions in context FlexibleSUSY`
     may shadow or be shadowed by other definitions.
~~~

**Comment by Expander** [comment]
> The `` Remove[Susyno`LieGroups`M] `` should be run only once per kernel.

If that works as well, please change it the way you propose! :)

**Comment by wkotlarski** [comment]
For me on a fresh clone of the repo I get
```
Creating C++ QFT class...
Launching kernels...

Part::pkspec1: The expression a cannot be used as a part specification.

Part::pkspec1: The expression a cannot be used as a part specification.

Part::pkspec1: The expression a cannot be used as a part specification.

General::stop: Further output of Part::pkspec1
     will be suppressed during this calculation.
From KernelObject[1, local]:
M::shdw: Symbol M appears in multiple contexts 
    {FlexibleSUSY`, Susyno`LieGroups`}; definitions in context FlexibleSUSY`
     may shadow or be shadowed by other definitions.
From KernelObject[2, local]:
M::shdw: Symbol M appears in multiple contexts 
    {FlexibleSUSY`, Susyno`LieGroups`}; definitions in context FlexibleSUSY`
     may shadow or be shadowed by other definitions.
From KernelObject[3, local]:
M::shdw: Symbol M appears in multiple contexts 
    {FlexibleSUSY`, Susyno`LieGroups`}; definitions in context FlexibleSUSY`
     may shadow or be shadowed by other definitions.
From KernelObject[4, local]:
M::shdw: Symbol M appears in multiple contexts 
    {FlexibleSUSY`, Susyno`LieGroups`}; definitions in context FlexibleSUSY`
     may shadow or be shadowed by other definitions.

The creation of C++ vertices took 4.8s
```

**Comment by Expander** [comment]
OK, I see. Just for my clarity: Is this the output from the generation of the code in `models/MSSM/`? I.e. it is not from the generation of the code in `test/`?

**Comment by wkotlarski** [comment]
Yes. Only I tested SM, not the MSSM.

**Comment by Expander** [comment]
Hmm, I cannot reproduce this issue on my machine. :/ Which version of Mathematica and SARAH are you using?

**Comment by wkotlarski** [comment]
I'm using Mathematica 11.2 and SARAH 4.14.3

**Comment by Expander** [comment]
I'm using 12.0.0, so maybe this is makes the difference?

**Comment by wkotlarski** [comment]
This is from a CI, where we use Wolfram Engine 12.1.0
```

Part::pkspec1: The expression a cannot be used as a part specification.

Part::pkspec1: The expression a cannot be used as a part specification.

General::stop: Further output of Part::pkspec1
     will be suppressed during this calculation.
      Two Ghost - One Vector Boson - Interactions
     Found 65 potential vertices. Calculating /65 . (
      (in progress)
      Two Ghost - One Scalar - Interactions
     Found 48 potential vertices. Calculating /48 . (
      (in progress)
From KernelObject[1, local]:
M::shdw: Symbol M appears in multiple contexts 
    {FlexibleSUSY`, Susyno`LieGroups`}; definitions in context FlexibleSUSY`
     may shadow or be shadowed by other definitions.
From KernelObject[2, local]:
M::shdw: Symbol M appears in multiple contexts 
    {FlexibleSUSY`, Susyno`LieGroups`}; definitions in context FlexibleSUSY`
     may shadow or be shadowed by other definitions.

The creation of C++ vertices took 7.1s
```

**Comment by Expander** [comment]
Would you mind trying the following patch?
~~~.patch
diff --git a/meta/Decays.m b/meta/Decays.m
index eb93e02a4..a30c45a99 100644
--- a/meta/Decays.m
+++ b/meta/Decays.m
@@ -2002,7 +2002,15 @@ CreateTotalAmplitudeSpecializations[particleDecays_List, modelName_] :=
            Print[""];
            FSFancyLine[];
            Print["Creating a C++ code for decay amplitudes..."];
-           ParallelEvaluate[(BeginPackage[#];EndPackage[];)& /@ contextsToDistribute, DistributedContexts->All];
+           ParallelEvaluate[
+               (BeginPackage[#];EndPackage[];
+                (* prevent shdw warning with Susyno`LieGroups`M: *)
+                Off[Remove::remal];
+                Remove[Susyno`LieGroups`M];
+                On[Remove::remal];
+               )& /@ contextsToDistribute,
+               DistributedContexts->All
+           ];
            specializations =
               AbsoluteTiming@ParallelMap[
                  (
~~~

**Comment by wkotlarski** [comment]
Small correction. We're fixing `meta/CXXDiagrams.m`, not `meta/Decays.m`. But in the end it doesn't help either.

**Comment by Expander** [comment]
Yes. I was just desperately trying to fix all the places where we use `ParallelMap`, but it does not help. :/

File: meta/CXXDiagrams.m [hunk 1]
```diff
@@ -1083,8 +1083,60 @@
...
+       Off[Remove::remal];
+       Remove[Susyno`LieGroups`M];
+       On[Remove::remal];
+      )& /@ contextsToDistribute,
+      DistributedContexts->Automatic
+   ];
+   (* without this CForm converts complex numbers using
+      Complex wrapper *)
+   ParallelEvaluate[
+      Unprotect[Complex];
```
Comment (uukhas) [inline meta/CXXDiagrams.m:1104]: This code exists inside `CConversion.m` already, so I think, that it should not be just copy-pasted here from there. I have found a safer way to do similar thing in #269.

File: meta/CXXDiagrams.m [hunk 2]
```diff
@@ -1159,54 +1159,62 @@
...
+         AbsoluteTiming@ParallelMap[
+            CreateVertex,
+            DeleteDuplicates[vertices], DistributedContexts->All
+         ],
+      cxxVertices =
+         AbsoluteTiming@Map[
+            CreateVertex,
+            DeleteDuplicates[vertices]
+         ],
+      Print["Error in CXXDiagrams. Variable FSEnableParallelism not defined."]; Quit[1];
```
Comment (uukhas) [inline meta/CXXDiagrams.m]: I suggest to use ``Utils`AssertOrQuit`` if possible.

**Pull request closed without merging**